packages/api/endpoints/granules.js

```diff
@@ -487,15 +488,32 @@ async function del(req, res) {
 async function get(req, res) {
   const {
     knex = await getKnexClient(),
+    collectionPgModel = new CollectionPgModel(),
   } = req.testContext || {};
   const { getRecoveryStatus } = req.query;
   const granuleId = req.params.granuleName;
+  const collectionId = req.params.collectionId;
+
   let granule;
+  let pgCollection;
   try {
-    granule = await getUniqueGranuleByGranuleId(knex, granuleId);
+    if (collectionId) {
+      pgCollection = await collectionPgModel.get(
+        knex, deconstructCollectionId(collectionId)
+      );
```

Comment:
Getting the collection here in the endpoint so that we can use the actual DB column `granule.collection_cumulus_id` going forward.